build-tools/packages/build-infrastructure/src/packageManagers.ts

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { IPackageManager, PackageManagerName } from "./types.js";
+
+export class PackageManager implements IPackageManager {
+	private constructor(public readonly name: PackageManagerName) {}
+
+	public installCommand(updateLockfile: boolean): string {
+		switch (this.name) {
+			case "npm": {
+				const command = "install";
+				const update = updateLockfile ? "--package-lock=true" : "--package-lock=false";
+				return `${command} ${update}`;
+			}
+
+			case "pnpm": {
+				const command = "install";
+				const update = updateLockfile ? "--no-frozen-lockfile" : "--frozen-lockfile";
+				return `${command} ${update}`;
+			}
+
+			case "yarn": {
+				return "install";
+			}
+
+			default: {
+				throw new Error(`Unknown package manager name: ${this.name}`);
+			}
+		}
+	}
+
+	public static load(name: PackageManagerName): IPackageManager {
+		return new PackageManager(name);
+	}
```

Comment:
Only see one use of this function outside this file. Should we just make the constructor public, get rid of load(), and have everything use createPackageManager()?

Reply:
I didn't want to make the constructor public to make it clear that you have to use the create/load functions, but the
duplication was confusing. I switched this to use the pattern of a static create method, but left the constructor private.

Reply:
Hmmm, that doesn't really work since the create function needs to be exported, and I don't want to export the class
itself. So back to public constructor and free function create.